share/spice/islamic_prayer_times/islamic_prayer_times.js

```diff
@@ -0,0 +1,100 @@
+(function (env) {
+    "use strict";
+
+    // As we are using UTC that does not change with the change of seasons,
+    // we need to check DST as local time or civil time may change if a time zone jurisdiction observes daylight saving time or summer time
+    function setDst(time, dst) {
+        return dst === 1 ? time.add(dst, 'hours') : time;
+    }
+
+    // Timezones can't be changed directly within the core moment.js library
+    // However, we can change timezones using the UTC offsets as the API provides it.
+    function tz(time, offset) {
+        var clone = time.clone();
+        return clone.utcOffset(offset).add(time.utcOffset() - clone.utcOffset(), 'minutes');
+    }
+
+    function getClosest(timings, offset, local) {
+        var date = local.format('YYYY-MM-DD');
+        if ('date_for' in timings) {
+            date = tz(moment.utc(timings.date_for + ' ' + offset, 'YYYY-MM-DD Z'), offset).format('YYYY-MM-DD');
+            delete timings.date_for;
+        }
+        for (var property in timings) {
+            if (timings.hasOwnProperty(property)) {
+                var time = tz(moment.utc(date + ' ' + timings[property] + ' ' + offset, 'YYYY-MM-DD hh:mm a Z'), offset);
+                if (local.diff(time) < 0) {
+                    var min = {
+                        time: time,
+                        diff: time.from(local),
+                        title: DDG.capitalize(property)
+                    };
+                    return min;
+                }
+            }
+        }
+        return false;
+    }
+
+    function getList(timings) {
+        var list = [];
+        for (var property in timings) {
+            if (timings.hasOwnProperty(property) && property !== "date_for") {
+                list[DDG.capitalize(property)] = timings[property].toUpperCase();
+            }
+        }
+        return list;
+    }
+
+    function setFormat(closest, index) {
+        return index === 0 ? closest.format('h:mm A') : closest.format('h:mm A, MMM D ') + '(Tomorrow)';
+    }
+
+    function setAddress(title, country_code, country, state) {
```

Comment:
Hmm looks like the API has some issues with Unicode...
![namaz_times_for_tokyo_at_duckduckgo](https://cloud.githubusercontent.com/assets/873785/9392318/08c04de4-474a-11e5-828b-6f38e526f108.png)

That's a tough case because the city, state, country are all blank and there's no way for use to know if the unicode is correct.